### ISSUE TITLE
A new LR scheduler in Pytorch

### ISSUE THREAD

**Issue by nikhil-iyer-97** [pytorch/pytorch#59556]
## 🚀 Feature
Proposal for a new LR Scheduler in Pytorch ( opening an issue before creating a PR )

Please refer to this paper for full details (https://arxiv.org/pdf/2003.03977.pdf). We compared our schedule against most of the inbuilt schedulers in torch.optim.lr_scheduler ( StepLR, MultiStepLR, CosineAnnealingLR , OneCycleLR) and have good results against all of them wrt generalization as well as wall clock time savings.

## Motivation

We proposed a new LR schedule in our paper by exploiting the loss landscape and designed the LR schedule to land in wider minima with a good probability. Wider minima are generally better for generalization as shown in multiple papers. Our LR scheduler is based on a hypothesis we proposed regarding the density of wide vs narrow minima. We show that our LR schedule generalizes better than most baseline schedules, as well as generalizing faster and saving a lot of GPU compute hours ( Please refer to Table 5,6 and 7 in the paper ). We have also added tables with results here (https://github.com/nikhil-iyer-97/wide-minima-density-hypothesis#main-results). 

P.S. We also show that we can push the generalization boundaries of many SOTA models just by modifying the LR schedule ( Please refer to section 5.3.6 in the paper ). 

## Pitch

Addition of the new proposed LR schedule to torch.optim.lr_scheduler. We have already created a simple framework for researchers to use our schedule here (https://github.com/nikhil-iyer-97/wide-minima-density-hypothesis#knee-lr-schedule).  I created an issue as I wanted to discuss this with the issue reviewers on how to take this forward. Please feel free to let me know what you think. All comments are welcome. 

## Additional context

Link to my repo ( https://github.com/nikhil-iyer-97/wide-minima-density-hypothesis ). We discuss our hypothesis and our lr schedule in brief and how we computed metrics to show that our lr schedule lands in a wider minima

cc @vincentqb @iramazanli

**Comment by iramazanli** [comment]
Thanks, @nikhil-iyer-97 for the proposal and for the research work. 

Indeed Explore-Exploit Learning Rate Scheduler seems to be interesting work. However, at this point, we decided to wait before adding the learning rate scheduler to PyTorch. 

We generally select algorithms/methods to add to PyTorch that is already widely used in the literature and we generally don't add the research work before this stage.

**Comment by nikhil-iyer-97** [comment]
Thanks for the discussion!
